### PR TITLE
fixed issue 60

### PR DESCRIPTION
**Pull request by johnnydevaprasad**
Fixed `z01-default_modules.csh` and `z01-default_modules.sh` as suggested.
Also fixed `z00_lmod.csh` to be aware of `USER_IS_ROOT`

### CODE REVIEW

File: Lmod/branches/7.0/z00_lmod.csh [hunk 1]
```diff
@@ -7,40 +7,40 @@
...
-
-    #
-    # If MANPATH is empty, Lmod is adding a trailing ":" so that
-    # the system MANPATH will be found
-    if ( ! $?MANPATH ) then
-      setenv MANPATH :
-    endif
-    setenv MANPATH `/usr/share/lmod/lmod/libexec/addto MANPATH /usr/share/lmod/lmod/share/man`
-
+if ( ! $?USER_IS_ROOT ) then
```
Comment (fgeorgatos) [inline Lmod/branches/7.0/z00_lmod.csh]: I think this would benefit a lot from using the same `&& exit` clause as the one of .sh, no?

Comment (fgeorgatos) [reply Lmod/branches/7.0/z00_lmod.csh]: ie.

```
if ( ! $?USER_IS_ROOT ) then
   exit
 endif
```

**Comment by fgeorgatos** [comment]
also: a bump in the .spec minor version?! (and corresponding filename change!)

File: Lmod/branches/7.0/z00_lmod.csh (hunk 1 above)
Comment (johnnydevaprasad) [inline Lmod/branches/7.0/z00_lmod.csh]: Do you mean:  use the same style as z01-default_modules.csh ?

**Comment by fgeorgatos** [comment]
1) include functionality of `$LMOD_DISABLE` in parallel to `$USER_IS_ROOT`
2) provide examples of `00-INIT-MODULES.*sh` or similar
3) add ./contrib
4) git ID in both `ml --version` and rpm name (and version bump, that helps too!)

**Comment by fgeorgatos** [comment]
@johnnydevaprasad : any progress here?!

File: Lmod/branches/7.0/z00_lmod.csh (hunk 1 above)
Comment (fgeorgatos) [reply Lmod/branches/7.0/z00_lmod.csh]: yes, exactly that

File: Lmod/branches/7.0/Lmod-illumina.spec [hunk 2]
```diff
@@ -10,10 +10,10 @@
...
 %define sles12      %(test -e /etc/os-release && source /etc/os-release && echo "${ID}  ${VERSION_ID}" | grep -q "^sles  12" && echo 1 || echo 0)
                 
-# %define git_rev     %(git rev-list --count --first-parent HEAD)
-%define git_rev     30
-%define git_tag     %(git describe --always)
-%define lmod_upstream_gitid git-079ce23
+%define git_rev     %(git rev-list --count --first-parent HEAD)
+# %define git_tag     %(git describe --always)
+%define git_tag     079ce23
+%define lmod_upstream_gitid git-%{git_tag}
```
Comment (fgeorgatos) [inline Lmod/branches/7.0/Lmod-illumina.spec:18]: would I be correct to point out that git_tag is undefined?

File: Lmod/branches/7.0/z01-default_modules.sh [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
+    else
+      if [ -f "/usr/share/modulefiles/StdEnv.lua" ];then
+        export LMOD_SYSTEM_DEFAULT_MODULES="StdEnv"
+      fi
+    fi
   fi
+  module --initial_load restore
+else
+  module refresh
 fi
```
Comment (fgeorgatos) [inline Lmod/branches/7.0/z01-default_modules.sh:17]: yeah, there's a party of nested IFs over here; T.B.D. let's give it one more pass for now...

File: Lmod/branches/7.0/Lmod-illumina.spec [hunk 4]
```diff
@@ -10,10 +10,10 @@
 %define sles11      %(test -e /etc/SuSE-release && test $(awk '$1=="VERSION"{printf $3}' /etc/SuSE-release) = "11" && echo 1 || echo 0)
 %define sles12      %(test -e /etc/os-release && source /etc/os-release && echo "${ID}  ${VERSION_ID}" | grep -q "^sles  12" && echo 1 || echo 0)
                 
-# %define git_rev     %(git rev-list --count --first-parent HEAD)
-%define git_rev     30
-%define git_tag     %(git describe --always)
-%define lmod_upstream_gitid git-079ce23
+%define git_rev     %(git rev-list --count --first-parent HEAD)
```
Comment (fgeorgatos) [inline Lmod/branches/7.0/Lmod-illumina.spec:16]: btw. did I mention that this is brilliant?! ;)

File: Lmod/branches/7.0/Lmod-illumina.spec (hunk 2 above)
Comment (johnnydevaprasad) [reply Lmod/branches/7.0/Lmod-illumina.spec:18]: it is defined in the next line. probably it is better to remove the commented out line.

**Comment by fgeorgatos** [comment]
we didn't implement (1) but this should not wait any longer, RPMs are needed super-ASAP; merging

**Pull request merged**